### PR TITLE
Use local venv for Makefile scripts

### PR DESCRIPTION
**Pull request by epcim**
I quite believe I had a problem with the loop I commented on update target.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -36,16 +37,23 @@ html:
...
+	@if ! which virtualenv; then echo "Please install virtualenv first";  exit 2; fi
+	@virtualenv -p python3 --no-site-packages $(PYENV_DIR)
+	@. $(PYENV_DIR)/bin/activate
+	$(PYENV_DIR)/bin/python -m pip install -r scripts/requirements.txt
 
 list: scripts_prerequisites
 	@echo $(FORMULAS)
 
-update_forks:
+update_forks: scripts_prerequisites
```
Comment (fpytloun) [inline Makefile:53]: Why update_forks is expected to install scripts requirements?

File: Makefile [hunk 2]
```diff
@@ -36,16 +37,23 @@ html:
 pdf:
 	make -C doc latexpdf
 
+PYENV_DIR=.virtualenv
 FORKED_FORMULAS_DIR=formulas
-FORMULAS=`python3 -c 'import sys; sys.path.append("scripts");from update_mrconfig import *; print(*get_org_repos(make_github_agent(), "salt-formulas"), sep="\n")'| egrep 'salt-formula-' | sed 's/salt-formula-//'`
+FORMULAS=`. $(PYENV_DIR)/bin/activate && python3 -c 'import sys; sys.path.append("scripts");from update_mrconfig import *; print(*get_org_repos(make_github_agent(), "salt-formulas"), sep="\n")'| egrep 'salt-formula-' | sed 's/salt-formula-//'`
```
Comment (fpytloun) [inline Makefile]: What if requirements are already installed on system and user does not need to care about virtualenv?

**Review by fpytloun — DISMISSED** [review]

**Comment by epcim** [comment]
in such case i had strange problem with pygithub class loading. i was able to solve it in isolated env. it does have just few deps so overhead is not big. On 19 Feb 2018 13:39, Filip Pytloun <notifications@github.com> wrote:@fpytloun requested changes on this pull request.



In Makefile:
>  FORKED_FORMULAS_DIR=formulas
-FORMULAS=`python3 -c 'import sys; sys.path.append("scripts");from update_mrconfig import *; print(*get_org_repos(make_github_agent(), "salt-formulas"), sep="\n")'| egrep 'salt-formula-' | sed 's/salt-formula-//'`
+FORMULAS=`. $(PYENV_DIR)/bin/activate && python3 -c 'import sys; sys.path.append("scripts");from update_mrconfig import *; print(*get_org_repos(make_github_agent(), "salt-formulas"), sep="\n")'| egrep 'salt-formula-' | sed 's/salt-formula-//'`

What if requirements are already installed on system and user does not need to care about virtualenv?

File: Makefile (hunk 1 above)
Comment (epcim) [reply Makefile:53]: its using virtualenv now, so it need's to run it each time just for sure..

File: Makefile (hunk 2 above)
Comment (epcim) [reply Makefile]: there is a problem with pygithub. Documented.Fail to load it's MainClas, I found it to fail when installing systemwide/virtualenv with linking system libraries.

**Comment by epcim** [comment]
Filip?

File: Makefile (hunk 2 above)
Comment (fpytloun) [reply Makefile]: pygithub package will be soon updated to latest version in Debian. Default should be to use system-wide stuff when available. Do not force user to use virtualenv.

File: Makefile [hunk 3]
```diff
@@ -79,5 +87,5 @@ remote_github_add:
      if ! git remote | grep $$ID 2>&1 > /dev/null ; then\
        git remote add $$ID git://github.com/$$ID/salt-formula-$$FORMULA;\
      fi;\
-     cd - > /dev/null;\
+     d - > /dev/null;\
```
Comment (fpytloun) [inline Makefile]: What's d?

**Comment by epcim** [comment]
Will redo with https://docs.pipenv.org/ and fix the "d"

**Pull request merged**